libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -73,8 +73,9 @@ class YulUtilFunctions
 
 	/// @returns a function that copies raw bytes of dynamic length from calldata
 	/// or memory to memory.
-	/// Pads with zeros and might write more than exactly length.
-	std::string copyToMemoryFunction(bool _fromCalldata);
+	/// @params _cleanup: If true, pads with zeros and might write more than
+	///         the exact length, else only the exact length will be written
+	std::string copyToMemoryFunction(bool _fromCalldata, bool _cleanup = true);
```

Comment:
I would remove the default. `true` is dangerous if you're not careful. Better if you have to make that decision explicitly. I would also give the parameter name that does not sound that harmless :)
```suggestion
	std::string copyToMemoryFunction(bool _fromCalldata, bool _unsafeCleanup);
```

As a side effect it would also let us see every use of this function in the review.